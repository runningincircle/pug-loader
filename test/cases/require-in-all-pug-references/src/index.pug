extends layout
include mixins

//- NamedBlock
block content
  //- Tag attributes
  img(src=require('./image.png') srcset=`${require('./image1.png')} 80w, ${require('./image2.png')} 90w`).a

  //- Code
  - var imageFile = require('./image.png');
  img(src=imageFile).b

  //- Mixin
  +image(require('./image.png'), 'logo-b').c

  //- Case, When
  - var num = 1;
  case num
    when 1
      img(src=require('./image1.png')).d1

  //- Conditional
  if true
    img(src=require('./image1.png')).e1
  if false
    //- do nothing
  else
    //- Conditional alternate
    img(src=require('./image2.png')).e2

  //- Each obj as array
  each img in [require('./image1.png'), require('./image2.png')]
    img(src=img).f1

  //- Each obj as json
  each img, key in { apple: require('./image1.png'), sony: require('./image2.png') }
    img(src=img alt=key).f2

  //- EachOf obj as Map, see feature: https://github.com/pugjs/pug/pull/3179
  each [key, img] of new Map([['apple', require('./image1.png')], ['sony', require('./image2.png')]])
    img(src=img alt=key).f3

  //- While
  - var n = 0;
  while n++ < 2
    img(src=require('./image.png') class=`g-${n}`)

  //- Mixin block
  +images-container
    img(src=require('./image1.png')).h1
    img(src=require('./image2.png')).h2

  //- Block
  include include-images

  //- Text, do nothing
  :code
    img(src=require('./image.png')).h