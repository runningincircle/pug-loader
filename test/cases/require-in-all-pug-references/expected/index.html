<html><head><link href="/styles.css" rel="stylesheet"><script src="/main.js"></script></head><body><p>Hello World!</p><img class="a" src="/assets/img/image.697ef306.png" srcset="/assets/img/image1.697ef306.png 80w, /assets/img/image2.697ef306.png 90w"><img class="b" src="/assets/img/image.697ef306.png"><img class="c" src="/assets/img/image.697ef306.png" alt="logo-b"><img class="d1" src="/assets/img/image1.697ef306.png"><img class="e1" src="/assets/img/image1.697ef306.png"><img class="e2" src="/assets/img/image2.697ef306.png"><img class="f1" src="/assets/img/image1.697ef306.png"><img class="f1" src="/assets/img/image2.697ef306.png"><img class="f2" src="/assets/img/image1.697ef306.png" alt="apple"><img class="f2" src="/assets/img/image2.697ef306.png" alt="sony"><img class="f3" src="/assets/img/image1.697ef306.png" alt="apple"><img class="f3" src="/assets/img/image2.697ef306.png" alt="sony"><img class="g-1" src="/assets/img/image.697ef306.png"><img class="g-2" src="/assets/img/image.697ef306.png"><div class="images"><img class="h1" src="/assets/img/image1.697ef306.png"><img class="h2" src="/assets/img/image2.697ef306.png"></div><img class="j1" src="/assets/img/image1.697ef306.png"><img class="j2" src="/assets/img/image2.697ef306.png"><code>img(src=require('./image.png')).h</code></body></html>